<a href="https://colab.research.google.com/github/eric999j/PILTest/blob/master/TestGoogleTranslate.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
!pip install SpeechRecognition
!pip install gTTS
!pip install pygame
!pip install googletrans # 直接檔案轉到這邊就可以了
!sudo apt-get install python3-pyaudio #嘗試用麥克風將聲音直接轉成文字輸出，則需要安裝 PyAudio 

     |████████████████████████████████| 32.8MB 124kB/s 
  Created wheel for gtts-token: filename=gTTS_token-1.1.3-cp36-none-any.whl size=4097 sha256=001124fd216e17163a1e4ff2b103b9f4efca68b812dbe464aed67b354860dc01
  Stored in directory: /root/.cache/pip/wheels/dd/11/61/33f7e51bf545e910552b2255eead2a7cd8ef54064b46dceb34
Successfully built gtts-token
     |████████████████████████████████| 11.4MB 3.4MB/s 
  Created wheel for googletrans: filename=googletrans-2.4.0-cp36-none-any.whl size=15776 sha256=c59263ac7ff1ae6d16aca771a4091bb1f158cc31ef28646365127007eb31e897
  Stored in directory: /root/.cache/pip/wheels/50/d6/e7/a8efd5f2427d5eb258070048718fa56ee5ac57fd6f53505f95
Successfully built googletrans
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libnvidia-common-430
Use 'sudo apt autoremove' to remove it.
The following additional packages will be installed:
  lib

In [11]:
from google.colab import drive  
drive.mount('/content/gdrive') 

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/gdrive


In [18]:
%cd /content/gdrive/'My Drive'/'Colab Notebooks'/
%ls

/content/gdrive/My Drive/Colab Notebooks
 0安裝Conda.ipynb                           Geth.ipynb
'1How to use gpu and colab.ipynb'           panda.mp3
'2TF2.0 Keras beginners.ipynb'              panda.wav
'3advanced tf2.ipynb'                       speech-emotion-recognition/
'Copy of ResNet.ipynb'                      TestGoogleTranslate2.ipynb
 DataFlair-DetectingFakeNews.ipynb          TestGoogleTranslate3.ipynb
 DataFlair-ObjectDetection.ipynb            TestGoogleTranslate.ipynb
 DataFlair-SpeechEmotionRecognition.ipynb   Tom-Chang-Deep-Lyrics.ipynb
'Drowsiness detection'/                     Tom-Chang-Deep-Lyrics-master/


In [0]:
import speech_recognition
r = speech_recognition.Recognizer()

In [0]:
with speech_recognition.AudioFile("panda.wav") as source: #wav才可以吃
     audio = r.record(source)

In [16]:
r.recognize_google(audio, language='zh-TW')

'猫熊'

https://shihnung.blogspot.com/2019/01/blog-post.html
最後打包聊天機器人為獨立執行檔

安裝 pyinstaller 套件 (此時版本為 3.4)

    conda install -c conda-forge pyinstaller

打包為單一執行檔 chatbot.exe

    pyinstaller -F chatbot.py
